plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloat.hs

```diff
@@ -437,7 +437,8 @@ An extreme alternative implementation is to treat *all strict* bindings as unflo
 -}
 hasNoEffects :: PLC.ToBuiltinMeaning uni fun => Binding tyname name uni fun a -> Bool
```

Comment:
I don't think this belongs here. It's not true that a type let has *effects*, it's just that you can't float it. So this should be in a separate predicate!